docs/src/apalache/config.md

```diff
@@ -1,23 +1,72 @@
-  - `out-dir` (String): The value of `out-dir` defines a path to the directory,
-  in which all Apalache runs write their outputs. Each run will produce a unique
-  subdirectory inside `out-dir` using the following convention:
-  `<SPECNAME>_yyyy-MM-dd_HH-mm-ss_<UNIQUEID>`. The use of `~` in the path
-  specification is permitted. The directory path need not already exist.
-  Example value: `'~/apalache-out'`.
-  If this value is not defined in `apalache-global-config.yml`, Apalache will, for each individual run, define it to be equal to `CWD/_apalache-out/`, where `CWD` is the current working directory.
-  - `profiling` (Bool): This flag governs the creation of `profile-rules.txt` used in [profiling](profiling.md). The file is only created if `profiling` is set to `True`.  Setting `profiling` to `False` is incompatible with the `--smtprof` flag.
-  The default is `False`.
-  - `write-intermediate` (Bool): This Boolean flag governs the creation of intermediate outputs. If set to `True`, apalache will produce an `intermediate` subdirectory in the run directory and output the state of the module after each pass.
-  The default is `False`.
-
-
-Example of a valid `apalache-global-config.yml`:
-```
-out-dir: '~/apalache-out'
-profiling: True
-write-intermediate: True
+# Apalache configuration
+
+Apalache supports configuration of some parameters governing its behavior.
+
+Application configuration is loaded from the following four sources:
+
+1. Command line arguments
+2. Environment variables
+3. A local configuration file
+4. The global configuration file
+
+The order of precedence of the sources follows their numbering: i.e., and any
+configuration set in an earlier numbered source overrides a configuration set in
+a later numbered source.
+
+## Command line arguments and environment variables
+
+To view the available command line arguments, run Apalache with the `--help`
+flag and consult the section on [Running the Too](./running.md) for more
+details.
+
+*Some* parameters configurable the command are also configurable via environment
+variables. These parameters are noted in the CLI's inline help. If a parameter
+is configured both through a CLI argument and an environment variable, then the
+CLI argument always takes precedence. 
+
+## Configuration files
+
+### File format and supported parameters
+
+Local configuration files support JSON and the JSON superset
+[HOCON](https://github.com/lightbend/config/blob/master/HOCON.md).
+
+Here's an example of a valid configuration for the currently 
+supported parameters, along with their default values: 
+
+```yaml
+# Directory in which to write all log files and records of each run
+out-dir: "${PWD}/_apalache-out"
+
+# Whether or not to write additional files, that report on intermediate
+# processing steps
+write-intermediate: false
```

Comment:
This example looks like yaml, not json, which contradicts the preceding sentence.

Reply:
Which sentence does it contradict? This is valid HOCON.